com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1422,67 +1406,76 @@ internal void HandleApproval(ulong ownerClientId, bool createPlayerObject, uint?
                     return;
                 }
 
-                // Inform old clients of the new player
-                foreach (KeyValuePair<ulong, NetworkClient> clientPair in ConnectedClients)
+                // Separating this into a contained function call for potential further future separation of when this notification is sent.
+                ApprovedPlayerSpawn(ownerClientId, playerPrefabHash ?? NetworkConfig.PlayerPrefab.GetComponent<NetworkObject>().GlobalObjectIdHash);
+            }
+            else
+            {
+                PendingClients.Remove(ownerClientId);
+                NetworkConfig.NetworkTransport.DisconnectRemoteClient(ownerClientId);
+            }
+        }
+
+        /// <summary>
+        /// Spawns the newly approved player
+        /// </summary>
+        /// <param name="clientId">new player client identifier</param>
+        /// <param name="playerPrefabHash">the prefab GlobalObjectIdHash value for this player</param>
+        internal void ApprovedPlayerSpawn(ulong clientId, uint playerPrefabHash)
+        {
+            foreach (KeyValuePair<ulong, NetworkClient> clientPair in ConnectedClients)
+            {
+                if (clientPair.Key == clientId ||
+                    clientPair.Key == ServerClientId || // Server already spawned it
+                    ConnectedClients[clientId].PlayerObject == null ||
+                    !ConnectedClients[clientId].PlayerObject.Observers.Contains(clientPair.Key))
                 {
-                    if (clientPair.Key == ownerClientId ||
-                        clientPair.Key == ServerClientId || // Server already spawned it
-                        ConnectedClients[ownerClientId].PlayerObject == null ||
-                        !ConnectedClients[ownerClientId].PlayerObject.Observers.Contains(clientPair.Key))
-                    {
-                        continue; //The new client.
-                    }
+                    continue; //The new client.
+                }
 
-                    var context = MessageQueueContainer.EnterInternalCommandContext(
-                        MessageQueueContainer.MessageType.CreateObject, NetworkChannel.Internal,
-                        new[] { clientPair.Key }, NetworkUpdateLoop.UpdateStage);
-                    if (context != null)
+                var context = MessageQueueContainer.EnterInternalCommandContext(MessageQueueContainer.MessageType.CreateObject, NetworkChannel.Internal,
+                    new[] { clientPair.Key }, NetworkUpdateLoop.UpdateStage);
+                if (context != null)
+                {
+                    using (var nonNullContext = (InternalCommandContext)context)
```

Comment:
I hate how nullables plague the codebase with null checks and yet another variable with seemingly very little benefits *sigh*...

Reply:
I agree the other issue is the implicit conversion is a function call. So `(InternalCommandContext)context` is actually calling Value property which is not always clear to the user and can also throw an exception if the value is not set so something to note about these casts.

Reply:
I was going to say outside of this PR, but it would be good to at least invert the if statements here so we bail early on error cases and to avoid indenting. I think this is one of the unfortunate side effects of a  lot of nullables.

Reply:
Yeah I hear you...but in my PR's defense it was that way pre-PR-955...I just moved that code into a separate method.  :)